components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java

```diff
@@ -384,42 +384,24 @@ private void updateFederatedAuthenticatorConfig(
             int authnId = getAuthenticatorIdentifier(dbConnection, idpId,
                     federatedAuthenticatorConfig.getName());
 
-            for (Property property : federatedAuthenticatorConfig.getProperties()) {
+            List<Property> singleValuedProperties = new ArrayList<>();
+            List<Property> multiValuedProperties = new ArrayList<>();
 
-                sqlStmt = IdPManagementConstants.SQLQueries.UPDATE_IDP_AUTH_PROP_SQL;
-                prepStmt2 = dbConnection.prepareStatement(sqlStmt);
-                prepStmt2.setString(1, property.getValue());
-                if (property.isConfidential()) {
-                    prepStmt2.setString(2, "1");
+            for (Property property : federatedAuthenticatorConfig.getProperties()) {
+                if (property.getName().contains(IdPManagementConstants.MULTI_VALUED_PROPERTY_CHARACTER)) {
```

Comment:
We can't check contains. _ can be in the middle of a name even. We have to check for pattern _[any number of digits]